src/Swarm/Game/Step.hs

```diff
@@ -2677,12 +2688,17 @@ updateRobotLocation ::
 updateRobotLocation oldLoc newLoc
   | oldLoc == newLoc = return ()
   | otherwise = do
+      newlocWithPortal <- applyPortal newLoc
       rid <- use robotID
       robotsByLocation . at oldLoc %= deleteOne rid
-      robotsByLocation . at newLoc . non Empty %= IS.insert rid
-      modify (unsafeSetRobotLocation newLoc)
+      robotsByLocation . at newlocWithPortal . non Empty %= IS.insert rid
+      modify (unsafeSetRobotLocation newlocWithPortal)
       flagRedraw
  where
+  applyPortal loc = do
+    lms <- use worldNavigation
+    return $ M.findWithDefault loc loc $ portals lms
```

Comment:
So am I understanding correctly that if you simply walk on top of a portal, you are instantly teleported?  What's the argument for doing it that way as opposed to, say, having a command like `travel` (or whatever) that activates a portal on which you are standing?  I'm honestly not sure which I would prefer, I can think of pros and cons of both.

Reply:
Ultimately I think we should support both :)

Teleporter-style, which is currently implemented, should be instantaneous.  But "ladders" and "stairs" would also be implemented as portals, and we could give the player the option to proceed down or up explicitly, after walking on to the cell.

To specify this, there will be an additional YAML attribute "direction" which can be, `up`, `down`, or omitted.